clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -156,6 +157,8 @@ public class SenderTest {
     private SenderMetricsRegistry senderMetricsRegistry = null;
     private final LogContext logContext = new LogContext();
 
+    private final Logger log = logContext.logger(SenderTest.class);
```

Comment:
I'm surprised that you added a logger when this massive test hasn't used one so far. Seems like logging is not required for working out whether the test passed or how it failed. I suggest removing it.

Reply:
Generally adding logging in tests isn't great. I would second removing it. I don't think the two lines added are worth the change. 

If you want that information about why a batch isn't being processed or if a batch is being processed you should add them as logs to the actual code. They can be trace if necessary

Reply:
Ok fair enough will remove the logging from tests.

I have already added traces in actual code, so that should be fine for now.